### PR TITLE
Add tlsConfig for ServiceMonitor

### PR DESCRIPTION
**Pull request by guicholeo**
Add configuration for `tlsConfig`

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] README is updated with new configuration values *(if applicable)* [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#documentation)
- [x] A PR is opened to update KEDA core ([repo](https://github.com/kedacore/keda)) *(if applicable, ie. when deployment manifests are modified)*

Fixes #

### CODE REVIEW

File: keda/Chart.yaml [hunk 1]
```diff
@@ -8,7 +8,8 @@ kubeVersion: ">=v1.23.0-0"
 
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
-version: 2.12.0
+
+version: 2.12.1
```
Comment (JorTurFer) [inline keda/Chart.yaml:12]: Please revert this change, we bump the version during releases

**Review by JorTurFer — COMMENTED** [review]
Could you rebase your PR and execute `helm-docs`?

File: keda/Chart.yaml (hunk 1 above)
Comment (guicholeo) [reply keda/Chart.yaml]: reverted

File: keda/templates/metrics-server/servicemonitor.yaml [hunk 2]
```diff
@@ -29,7 +29,7 @@ spec:
   {{- toYaml . | nindent 4 }}
   {{- end }}
   endpoints:
-  - port: {{ .Values.prometheus.metricServer.portName }}
+  - port: {{ .Values.prometheus.metricServer.serviceMonitor.port }}
```
Comment (JorTurFer) [inline keda/templates/metrics-server/servicemonitor.yaml:32]: Is this a breaking change? We should avoid them

Comment (guicholeo) [reply keda/templates/metrics-server/servicemonitor.yaml:32]: It will be if users are overwriting the `portName` to be different than `serviceMonitor.port` otherwise, it should be fine since both ports need to be the same, and changing to `serviceMonitor.port` follows the same configuration from `servicemonitor.yaml` I can add a check to make sure the port names are the same as well

Comment (JorTurFer) [reply keda/templates/metrics-server/servicemonitor.yaml:32]: > It will be if users are overwriting the portName to be different than serviceMonitor.port

so, it's a breaking change indeed xD We need to ensure backward compatibility

Comment (guicholeo) [reply keda/templates/metrics-server/servicemonitor.yaml]: I removed the changes

**Review by JorTurFer — APPROVED** [review]
Thanks for the contruibution!

**Comment by JorTurFer** [comment]
I broke the PR trying to fix the conflict for merging it, as we want to cut a release tomorrow, I'm going to open a PR with your changes adding you as co-author @guicholeo , sorry :(
